src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -951,14 +951,55 @@ private void buildDockerImage(File tempDockerfile, File userDockerfile) throws P
             debug("Docker build context: " + userDockerfile.getParent());
             buildCmd = "docker build -f " + tempDockerfile + " -t " + imageName + " " + userDockerfile.getParent();
             info(buildCmd);
-            String buildOutput = execDockerCmd(buildCmd, dockerBuildTimeout);
-            debug("Docker build output: " + buildOutput);
+            long startTime = System.currentTimeMillis();
+            runCommandAndLog(getRunProcess(buildCmd), dockerBuildTimeout);
+            checkDockerIgnore(startTime, userDockerfile);
+            info("Completed building Docker image.");
+        } catch (IllegalThreadStateException  e) {
+            // the timeout was too short and the docker command has not yet completed.
+            debug("IllegalThreadStateException, message="+e.getMessage());
+            error("The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds. " +
+                "Use the dockerBuildTimeout option to specify a longer period or " +
+                "add files not needed in the container to the .dockerignore file.", e);
+            throw new PluginExecutionException("Could not build Docker image using Dockerfile: " + 
+                userDockerfile.getAbsolutePath() + ". Address the following docker build error and then start dev mode again: " +
+                "The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds", e);
```

Comment:
I think we should remove the `Address the following docker build error and then start dev mode again:` piece, because we are not showing the docker build error here.

Reply:
Good point. If anyone were to ask why we are duplicating all these messages here I would respond it is because we want to tune the response to the situation. In this case a docker build. I'll take a look.

Reply:
Since we are throwing PluginExecutionException dev mode will exit so the error message is not necessary. We'll just put the message in the exception.